source/agora/node/Runner.d

```diff
@@ -133,31 +133,42 @@ public Listeners runNode (Config config)
     auto tls_ctx = getTLSContext();
     if (result.admin !is null)
     {
-        if (!tls_ctx)
-            throw new Exception("This node has admin interface enabled so should be configured for ssl");
 
         bool checkPassword (string user, string password) @safe
         {
             return user == config.admin.username && password == config.admin.pwd;
         }
-        log.info("Admin interface listening will be on {}:{}", config.admin.address, config.admin.port);
         auto adminrouter = new URLRouter();
         adminrouter.any("*", performBasicAuth("Agora Admin", &checkPassword));
         adminrouter.registerRestInterface(result.admin);
         auto settings = new HTTPServerSettings(config.admin.address);
         settings.port = config.admin.port;
-        settings.tlsContext = tls_ctx;
+        if (config.admin.tls)
+        {
+            if (!tls_ctx)
+                throw new Exception("This node has tls enforced for admin so should be configured for ssl");
```

Comment:
A minor point, but here the message doesn't make much sense to the user IMO.
"enforced" implies that there's a way to have it not enforced, meaning one could use http and https side by side, however that is not the case.
Additionally, a good error message should not only tell the user what is wrong, but how to fix things, if there is an unambiguous way to do so. Think about the "did you mean XXX" [other](https://github.com/dlang/dmd/search?q=errorSupplemental) supplemental [messages](https://github.com/dlang/dmd/search?q=deprecationSupplemental).

In other cases, I would say that the following would be a decent error message:
```
Either add TLS certificates to $LOCATION, set `admin.tls` in $CONFIG_PATH to `false`, or use `-O admin.tls=false` as command line argument.
```

As it provides all 3 options, with the preferred option being listed in order (actually use TLS, disable via config, or disable temporarily). But since this is TLS, and people tend to bypass security when it's not convenient, we should restrict ourselves to remind them of what files are looked up and where they are looked up (path, and key / pem).

I think that's better handled by passing a `bool required` argument to `getTLSContext`, so let's leave it for another PR.

For another example, see this:
https://github.com/bosagora/agora/blob/d0de2335ea2fc6004b1fdceabf0975769b3c224f/source/agora/node/admin/Setup.d#L115-L118

This case is a bit different though, as the user had to explicitly provide `--initialize=ARG` on the CLI, so suggesting to not use that argument is obviously not a solution.

Reply:
Updated to log more info for end user.